src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt

```diff
@@ -222,6 +222,19 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase() {
             }
     """)
 
+    fun testInvalidChainComparison() = checkErrors("""
+        fn foo(x: i32) {
+            <error descr="Chained comparison operator require parentheses">1 < x < 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 > x < 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 > x > 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 < x > 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 <= x < 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 < x <= 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 == x < 3</error>;
+            <error descr="Chained comparison operator require parentheses">1 < x == 3</error>;
+        }
```

Comment:
Could you add a couple of "true negative" cases, like `10 < x && x < 20` and `(a == b) == c`?